client-local/src/main/java/com/cloudera/livy/client/local/LocalClient.java

```diff
@@ -248,76 +251,61 @@ public void run() {
       // mode, the driver options need to be passed directly on the command line. Otherwise,
       // SparkSubmit will take care of that for us.
       String master = conf.get("spark.master");
+      LOG.info("Master : " + master);
       Preconditions.checkArgument(master != null, "spark.master is not defined.");
 
-      List<String> argv = Lists.newArrayList();
-
-      if (sparkHome != null) {
-        argv.add(new File(sparkHome, "bin/spark-submit").getAbsolutePath());
-      } else {
-        LOG.info("No spark.home provided, calling SparkSubmit directly.");
-        argv.add(new File(System.getProperty("java.home"), "bin/java").getAbsolutePath());
-
         if (master.startsWith("local") ||
               master.startsWith("mesos") ||
               master.endsWith("-client") ||
               master.startsWith("spark")) {
+          launcher.setMaster(master);
```

Comment:
Why is this inside the condition?

If your master is "yarn" or "yarn-cluster", for example, you will not set a master and things will not work the way the user expects.